index.js

```diff
@@ -91,453 +76,471 @@ function serveIndex(root, options) {
     throw new TypeError('serveIndex() root path required');
   }
 
+  var pathLib = opts.path || nativePathLib;
+  var normalize = pathLib.normalize
+  , extname = pathLib.extname
+  , sep = pathLib.sep
+  , join = pathLib.join
+  , resolve = pathLib.resolve;
+
+  // default Stylesheet
+  var defaultStylesheet = join(__dirname, 'public', 'style.css');
+  // default Template
+  var defaultTemplate = join(__dirname, 'public', 'directory.html');
+
+
   // resolve root to absolute and normalize
   var rootPath = normalize(resolve(root) + sep);
 
   var filter = opts.filter;
   var hidden = opts.hidden;
-  var icons = opts.icons;
+  var showIcons = opts.icons;
   var stylesheet = opts.stylesheet || defaultStylesheet;
   var template = opts.template || defaultTemplate;
   var view = opts.view || 'tiles';
   var filesystem = opts.fs || fs;
-
-  return function (req, res, next) {
-    if (req.method !== 'GET' && req.method !== 'HEAD') {
-      res.statusCode = 'OPTIONS' === req.method ? 200 : 405;
-      res.setHeader('Allow', 'GET, HEAD, OPTIONS');
-      res.setHeader('Content-Length', '0');
-      res.end();
-      return;
-    }
-
-    // parse URLs
-    var url = parseUrl(req);
-    var originalUrl = parseUrl.original(req);
-    var dir = decodeURIComponent(url.pathname);
-    var originalDir = decodeURIComponent(originalUrl.pathname);
-
-    // join / normalize from root dir
-    var path = normalize(join(rootPath, dir));
-
-    // null byte(s), bad request
-    if (~path.indexOf('\0')) return next(createError(400));
-
-    // malicious path
-    if ((path + sep).substr(0, rootPath.length) !== rootPath) {
-      debug('malicious path "%s"', path);
-      return next(createError(403));
+  var responseHandler = opts.responseHandler || null;
+
+  /**
+   * Respond with text/html.
+   */
+  function html(req, res, files, next, dir, showUp, displayIcons, path, view, template, stylesheet, filesystem) {
```

Comment:
nit: moving so much code around cause a huge diff.  If possible, please move the code to end so original code at line 105 remains where they were to help with reduce code changes.

Reply:
named the returning function and retained it as per the original order.
moved the return statement to the end of the function.
hope this reduces the difference considerably.